plugins/aggregators/valuecounter/valuecounter.go

```diff
@@ -0,0 +1,94 @@
+package valuecounter
+
+import (
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+)
+
+type aggregate struct {
+	name       string
+	tags       map[string]string
+	fieldCount map[string]int
+}
+
+type ValueCounter struct {
+	cache       map[uint64]aggregate
+	Field_names []string
+}
+
+func NewValueCounter() telegraf.Aggregator {
+	vc := &ValueCounter{}
+	vc.Reset()
+	return vc
+}
+
+var sampleConfig = `
+  ## General Aggregator Arguments:
+  ## The period on which to flush & clear the aggregator.
+  period = "30s"
+  ## If true, the original metric will be dropped by the
+  ## aggregator and will not get sent to the output plugins.
+  drop_original = false
+  ## The fields for which the values will be counted
+  field_names = ""
+`
+
+func (vc *ValueCounter) SampleConfig() string {
+	return sampleConfig
+}
+
+func (vc *ValueCounter) Description() string {
+	return "Count the occurance of values in fields."
+}
+
+// Run on every metric which passes the plugin
+func (vc *ValueCounter) Add(in telegraf.Metric) {
+	id := in.HashID()
+
+	// Check if the cache already has an entry for this metric, if not create it
+	if _, ok := vc.cache[id]; !ok {
+		a := aggregate{
+			name:       in.Name(),
+			tags:       in.Tags(),
+			fieldCount: make(map[string]int),
+		}
+		vc.cache[id] = a
+	}
+
+	// Check if this metric has fields which we need to count, if so increment
+	// the count.
+	for fk, fv := range in.Fields() {
```

Comment:
I wonder if we should skip floats?  One thing that would be bad is if there is a frequently changing float value, which could result in many new fields.

If we don't disallow this, maybe we should add a warning in the README.

Reply:
good suggestion, will check if I can implement this